core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -154,8 +154,21 @@ public void start() throws IOException, IllegalStateException {
 
                     frameMat.copyTo(frameOutputSocket.getValue().get());
                     frameOutputSocket.setValue(frameOutputSocket.getValue().get());
-                    long thisMoment = System.currentTimeMillis();
-                    frameRateOutputSocket.setValue(1000 / (thisMoment - lastFrame));
+
+                    final long thisMoment = System.nanoTime();
+                    final long elapsedTime = thisMoment - lastFrame;
+                    if (elapsedTime == 0) {
+                        /* This can happen when using system.currentTimeMillis():
+                         * https://github.com/WPIRoboticsProjects/GRIP/issues/199
+                         * As such we use system.nanoTime().
+                         * However, we add an assertion to ensure this bug doesn't reappear.
+                         */
+                        assert elapsedTime != 0 : "The time between grabbing two frames was less than 1 nanosecond";
```

Comment:
Why couldn't it occur with `System.nanoTime()`?  Monotonically increasing doesn't mean every call is greater than every previous call, it just never goes backwards.

This is also not an appropriate place to use an assertion, since we'd only see it during development.  So, if we're asserting something, it's a test to make sure it'll be like that before we ship with.  Since the thing you're asserting is a timing condition, passing one time does not mean it will always pass.